### ISSUE TITLE
Quick Fix Option Not Appearing for Custom Stylelint Rule in VSCode

### ISSUE THREAD

**Issue by mcantonbul** [stylelint/stylelint#7955]
### What is the problem you're trying to solve?

Hi there,

I'm encountering an issue with a custom Stylelint rule that I have developed. While the rule's --fix option works correctly when run via the command line (npx stylelint '**/*.scss' --fix), the quick fix option does not appear in VSCode's Stylelint extension when hovering over the issues or using the context menu.

**Custom Rule:**
My custom Stylelint rule is designed to enforce the use of predefined SCSS variables for font sizes instead of hardcoded values. The rule includes a fix function to automatically replace these values.

Here's a simplified version of the rule:

```javascript
const stylelint = require('stylelint');
const ruleName = 'font-size-variable';
const messages = stylelint.utils.ruleMessages(ruleName, {
  expected: (value) => `Expected font-size to use a variable instead of ${value}`
});

const rule = (primaryOption) => {
  return (root, result) => {
    root.walkDecls('font-size', (decl) => {
      const value = decl.value;
      let fix;

      if (value === '1rem') {
        fix = () => {
          decl.value = '$font-size-m';
        };
        stylelint.utils.report({
          result,
          ruleName,
          node: decl,
          message: messages.expected(value),
          fix
        });
      } else if (value === '2rem') {
        fix = () => {
          decl.value = '$font-size-l';
        };
        stylelint.utils.report({
          result,
          ruleName,
          node: decl,
          message: messages.expected(value),
          fix
        });
      }
    });
  };
};

rule.meta = {
  fixable: 'code',
};

module.exports = stylelint.createPlugin(ruleName, rule);
module.exports.ruleName = ruleName;
module.exports.messages = messages;
```
**Issue:**
The custom rule's fixes work as expected when using --fix from the command line, but the quick fix option does not appear in VSCode when manually interacting with the issues reported by Stylelint.

**Steps Taken**
VSCode Configuration:

Verified that editor.codeActionsOnSave and source.fixAll.stylelint are properly set in settings.json.
Ensured that the Stylelint extension is up-to-date.

**Meta Configuration:**
Checked that rule.meta includes the fixable property set to 'code'.

### What solution would you like to see?

Is there a specific configuration or additional setup required to make quick fix options available for custom Stylelint rules in VSCode?
Are there any known issues or limitations with Stylelint's quick fix functionality that could explain this behavior?
I would appreciate any guidance or suggestions on how to resolve this issue.

Thank you!

**Comment by Mouvedia** [comment]
@mcantonbul you can refer yourself to the discussion starting with [this comment](https://github.com/stylelint/stylelint/issues/4664#issuecomment-2184137275).

I am of the opinion, that the feature that you are requesting could be provided if we had a `--dry-run` flag that would have a specific output.
e.g. something like `Warning & { fix?: { replacement: string, range: string } }` as decided in [this comment](https://github.com/stylelint/stylelint/issues/7788#issuecomment-2186850056)

In short the extension would always attempt `--fix --dry-run` and store that data after the lint so that a quick fix can be provided and be immediate once requested. It should be behind an option though because fixing is slightly slower than just only linting.

TL;DR: a mutation-free `--fix` (doesn't modify the sources targeted)

---

@ybiquitous Can we pin this issue?

**Comment by romainmenke** [comment]
@mcantonbul Is the request behavior something that works with other Stylelint rules?
Or something that doesn't work at all for Stylelint?

I wasn't able to infer that from your question.

-----

@Mouvedia I don't think we need to pin this issue.
We already have enough issues tracking improvements around autofixes and replacement ranges/text :)

**Comment by Mouvedia** [comment]
> @Mouvedia I don't think we need to pin this issue. We already have enough issues tracking improvements around autofixes and replacement ranges/text :)

I didn't mean as of right now.
But a _Add `--fix-dry-run` flag_ issue should at least have the _priority: high_ label.

**Comment by mcantonbul** [comment]
> @mcantonbul Is the request behavior something that works with other Stylelint rules? Or something that doesn't work at all for Stylelint?
> 
> I wasn't able to infer that from your question.
> 
> @Mouvedia I don't think we need to pin this issue. We already have enough issues tracking improvements around autofixes and replacement ranges/text :)

Thank you for your response.

It seems I mistakenly opened this issue in the main Stylelint repository instead of the Stylelint extension repository for VSCode.

The issue I am facing is related to the VSCode extension's quick fix functionality, which is not appearing for custom Stylelint rules, even though it works fine with other rules and through the command line.

I will open a new issue in the Stylelint extension repository to seek assistance with this matter. Thank you for your understanding!

Best regards,
